### PR TITLE
feat(chip): Stage 4 Leading Icon

### PR DESCRIPTION
**Pull request by andresin87**
**TASK**: 

### Types of changes

- [x] ✨ New feature (non-breaking change which adds functionality)

### CODE REVIEW

**Comment by andresin87** [comment]
🙏

File: packages/components/chip/src/Chip.stories.tsx [hunk 1]
```diff
@@ -29,6 +31,11 @@ export const Default: StoryFn = _args => (
   <div className="flex flex-col items-start gap-md">
     {designs.map(design => (
       <Chip design={design} key={design}>
+        <Chip.LeadingIcon>
```
Comment (andresz1) [inline packages/components/chip/src/Chip.stories.tsx:34]: I think that would be better to create a dedicated story for this

File: packages/components/chip/src/ChipLeadingIcon.tsx [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import React, { ComponentPropsWithoutRef, forwardRef } from 'react'
+
+export type ChipLeadingIconProps = ComponentPropsWithoutRef<'span'>
+export const ChipLeadingIcon = forwardRef<HTMLSpanElement, ComponentPropsWithoutRef<'span'>>(
+  ({ children }) => {
+    return <span className="mr-sm flex h-full items-center justify-center">{children}</span>
```
Comment (andresz1) [inline packages/components/chip/src/ChipLeadingIcon.tsx:6]: `className` support ?

File: packages/components/chip/src/ChipContent.tsx [hunk 3]
```diff
@@ -1,9 +1,8 @@
 import React, { ComponentPropsWithoutRef, forwardRef } from 'react'
 
-export const ChipContent = forwardRef<HTMLSpanElement, ComponentPropsWithoutRef<'span'>>(
-  ({ children }) => {
-    return <span className="inline-block grow truncate">{children}</span>
-  }
-)
+export type ChipContentProps = ComponentPropsWithoutRef<'span'>
+export const ChipContent = forwardRef<HTMLSpanElement, ChipContentProps>(({ children }) => {
+  return <span className="inline-block grow truncate">{children}</span>
```
Comment (andresz1) [inline packages/components/chip/src/ChipContent.tsx:5]: `className` support ?

File: packages/components/chip/src/Chip.stories.tsx (hunk 1 above)
Comment (andresin87) [reply packages/components/chip/src/Chip.stories.tsx:34]: stories will be fully created on stage 5

File: packages/components/chip/src/index.ts [hunk 4]
```diff
@@ -3,12 +3,19 @@ import { FC } from 'react'
 import { Chip as Root, ChipProps } from './Chip'
 import { ChipClearButton } from './ChipClearButton'
 import { ChipContent } from './ChipContent'
+import { ChipLeadingIcon } from './ChipLeadingIcon'
 
 export { type ChipClearButtonProps } from './ChipClearButton'
 export { type ChipContentProps } from './ChipContent'
+export { type ChipLeadingIconProps } from './ChipLeadingIcon'
 
 ChipClearButton.displayName = 'Chip.ClearButton'
```
Comment (Powerplex) [inline packages/components/chip/src/index.ts:12]: There is no test env deployed so I can't test, but I think without displayNames for LeadingIcon and other subComponents, markup will be wrong in the "show source" of storybook.

Comment (andresin87) [reply packages/components/chip/src/index.ts:12]: for sure stories is on stage 5

Comment (andresin87) [reply packages/components/chip/src/index.ts:12]: test stage 6

**Review by kikoruiz — APPROVED** [review]

**Review by Powerplex — APPROVED** [review]

**Pull request merged**